### DISCUSSION TITLE
Tooltip following cursor position on hover

### DISCUSSION THREAD

**Discussion by Rednegniw** [radix-ui/primitives#1090] [Ideas]
Hi, first, amazing library! So thank you very much for developing it. 

Just had an idea - it would be cool if there was a component such as a tooltip that would follow the mouse/cursor position while hovering over an element. It might interesting, for example, when you want to tell the user why they can't click on a disabled element and so on.

Something like this - https://codepen.io/paulvddool/pen/mROEGa (not mine, found randomly on google)

**Comment by Rednegniw** [comment]
Just bumping this to get an answer. Is there a possibility of adding a follow-cursor on hover tooltip? Thanks

**Reply by jjenzz** [reply]
I've converted this to an issue as a feature request https://github.com/radix-ui/primitives/issues/1323

**Reply by Rednegniw** [reply]
Thank you 🙂

**Comment by giovanniincammicia** [comment]
Hi everyone, while we wait for a primitive, I've implemented it with the Tooltip component. Hope this helps.
The idea is to use as a Trigger a 0x0 div that just follows the cursor position and then play with onMouseEnter and onMouseLeave properties on the hoverable div.
https://codesandbox.io/s/inspiring-tdd-2m0vkj
@Rednegniw @jjenzz

**Reply by benoitgrelard** [reply]
This is nice but unfortunately not accessible with keyboard.

**Reply by giovanniincammicia** [reply]
That's true :( it's definitely a workaround waiting for the official solution

**Reply by Rednegniw** [reply]
Hi! This seems to no longer work after updating to Tooltip 1.0.0. Do you know what should be changed to achieve similar results? I can't seem to figure it out :/ @GiovanniIncammicia

**Reply by giovanniincammicia** [reply]
Hi @Rednegniw , sorry for the late reply, I haven't been able to upgrade yet, I'll try as soon as possible

**Reply by Rednegniw** [reply]
Thank you @GiovanniIncammicia ! Looking forward to it.

**Reply by ZakRabe** [reply]
With 1.0 you can add a `key` to `Tooltip.Content` based on the mouse position. and it should redraw the content following the mouse

**Comment by franklinjavier** [comment]
I created a workaround using a custom hook to retrieve the mouse coordinates and assign them to the `offsets` props.

https://codesandbox.io/p/sandbox/goofy-star-29mzkc?file=/src/App.tsx

**Reply by heecheon92** [reply]
such a brilliant work. THANK YOU!

**Reply by raphaelobene** [reply]
Thank you kind sir...this is perfect.

**Reply by faribauc** [reply]
This is great!

**Reply by leo-fidlabs** [reply]
king, ty 🙏

**Reply by therealtgd** [reply]
Thank you!
